.github/workflows/vsbuild_xp.yml

```diff
@@ -96,3 +99,118 @@ jobs:
+          if (-not(Test-Path -Path src\dosbox-x.exe -PathType Leaf)) {exit 1}
+          #top=`pwd`
+          strip -s src/dosbox-x.exe
+          mkdir -p mingw-bin
+          cp src/dosbox-x.exe mingw-bin/dosbox-x_mingw_lowend_SDL2.exe
+      - name: Build MinGW32 lowend SDL1
+        shell: pwsh
+        run: |
+          make clean
+          mingw-get\msys\1.0\msys.bat /mingw/msys/1.0/runbuild.sh build-mingw-lowend
+          if (-not(Test-Path -Path src\dosbox-x.exe -PathType Leaf)) {exit 1}
+          #top=`pwd`
+          strip -s src/dosbox-x.exe
+          cp src/dosbox-x.exe mingw-bin/dosbox-x_mingw_lowend_SDL1.exe
+      - name: Cache MinGW32 lowend builds
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ github.workspace }}/mingw-bin
+          key: mingw-bin-${{ github.sha }}
+  build-XP-installer:
+    permissions:
+      actions: write  # for styfle/cancel-workflow-action to cancel/stop running workflows
+      contents: write # for actions/checkout to fetch code and softprops/action-gh-release
+    runs-on: windows-latest
+    needs: [MSBuild_XP_CI_build, MinGW32_lowend_CI_build]
+    env:
+      GH_TOKEN: ${{ github.token }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Restore Visual Studio builds
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ github.workspace }}/vs-bin
+          key: vs-bin-${{ github.sha }}
+      - name: Restore MinGW builds
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ github.workspace }}/mingw-bin
+          key: mingw-bin-${{ github.sha }}
+      - name: Prepare files
+        shell: bash
+        run: |
+          #DOSBOX_X_RELEASE=`gh release list -L1 | grep -o "....-..-.." | head -n 1 | sed -e "s/-/./g"`
+          #echo "DOSBOX_X_RELEASE=$DOSBOX_X_RELEASE" >> $GITHUB_ENV
+          echo "timestamp=`date +%F-%T | sed -e 's/:/-/g' | sed -e 's/-//g'`" >> $GITHUB_ENV
+          cp dosbox-*.conf contrib/windows/installer/
+          #ls -lg vs-bin
+          #ls -lg mingw-bin
+          mkdir -p contrib/windows/installer/WinXP
+          cp vs-bin/*.exe contrib/windows/installer/WinXP
+          cp mingw-bin/*.exe contrib/windows/installer/WinXP
+          #ls -lg contrib/windows/installer/WinXP
+          cd contrib/windows/installer/
+          ./ISCC.exe ./DOSBox-X-setupXP.iss
+      - name: Upload preview installer
+        uses: actions/upload-artifact@v3.1.2
+        with:
+          name: dosbox-x-winXP-${{ env.timestamp }}-setup.exe
+          path: contrib/windows/installer/dosbox-x-winXP-*.exe
+      - name: Upload XP build installer release version
+        uses: softprops/action-gh-release@v1
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          files: |
+            contrib/windows/installer/dosbox-x-winXP-*.exe
+      - name: Clean cache
```

Comment:
I don't get it. Caches are for data that can be repurposed between multiple runs of workflows. You can download artifacts from other jobs with [`actions/download-artifacts`](https://github.com/actions/download-artifacts).